src/Controls/tests/Core.UnitTests/TypedBindingFromExpressionUnitTests.cs

```diff
@@ -0,0 +1,1704 @@
+		}
+
+		[Fact, Category("[Binding] Complex paths")]
+		[Description("When part of a complex path can not be evaluated during an update, bindables should return to their default value.")]
+		public void NullInPathUsesDefaultValue()
+		{
+			var vm = new ComplexMockViewModel
+			{
+				Model = new ComplexMockViewModel()
+			};
+
+			var property = BindableProperty.Create("Text", typeof(string), typeof(MockBindable), "foo bar");
+
+			var bindable = new MockBindable();
+			bindable.SetBinding(property, static (ComplexMockViewModel cvm) => cvm.Model.Text, mode: BindingMode.OneWay);
+			bindable.BindingContext = vm;
+
+			vm.Model = null;
+
+			Assert.Equal(property.DefaultValue, bindable.GetValue(property));
+			Assert.True(MockApplication.MockLogger.Messages.Count == 0,
+				"An error was logged: " + MockApplication.MockLogger.Messages.FirstOrDefault());
+		}
+
+		[Fact, Category("[Binding] Complex paths")]
+		[Description("When part of a complex path can not be evaluated during an update, bindables should return to their default value.")]
+		public void NullContextUsesDefaultValue()
+		{
+			var vm = new ComplexMockViewModel
+			{
+				Model = new ComplexMockViewModel
+				{
+					Text = "vm value"
+				}
+			};
+
+			var property = BindableProperty.Create("Text", typeof(string), typeof(MockBindable), "foo bar");
+			
+			var bindable = new MockBindable();
+			bindable.SetBinding(property, static (ComplexMockViewModel cvm) => cvm.Model.Text, mode: BindingMode.OneWay);
+			bindable.BindingContext = vm;
+
+			Assert.Equal(bindable.GetValue(property), vm.Model.Text);
+
+			bindable.BindingContext = null;
+
+			Assert.Equal(property.DefaultValue, bindable.GetValue(property));
+			Assert.True(MockApplication.MockLogger.Messages.Count == 0,
+				"An error was logged: " + MockApplication.MockLogger.Messages.FirstOrDefault());
+		}
+
+		[Fact, Category("[Binding] Complex paths")]
+		[Description("When part of a complex path can not be evaluated during an update, bindables should return to their default value, or TargetNullValue")]
+		public void NullContextUsesFallbackValue()
+		{
+			var vm = new ComplexMockViewModel
+			{
+				Model = new ComplexMockViewModel
+				{
+					Text = "vm value"
+				}
+			};
+
+			var property = BindableProperty.Create("Text", typeof(string), typeof(MockBindable), "foo bar");
+			var bindable = new MockBindable();
+			bindable.SetBinding(property, static (ComplexMockViewModel cvm) => cvm.Model.Text, mode: BindingMode.OneWay, fallbackValue: "fallback");
+			bindable.BindingContext = vm;
+
+			Assert.Equal(bindable.GetValue(property), vm.Model.Text);
+
+			bindable.BindingContext = null;
+
+			Assert.Equal("fallback", bindable.GetValue(property));
+			Assert.True(MockApplication.MockLogger.Messages.Count == 0,
+				"An error was logged: " + MockApplication.MockLogger.Messages.FirstOrDefault());
+		}
+
+		[Fact]
+		//https://github.com/xamarin/Microsoft.Maui.Controls/issues/4103
```

Comment:
nit^2: 
I saw the comment that these tests are coming from another file, just wanted to draw the attention that these:
`https://github.com/xamarin/Microsoft.Maui.Controls/issues/*` links are broken. Not sure what was the original URL, just wondering if we should remove them.

Reply:
This is probably a bad find/replace, I think it's a Xamarin.Forms link?
```
https://github.com/xamarin/Xamarin.Forms/issues/4103
```